appbuilder/core/components/llms/base.py

```diff
@@ -110,8 +110,29 @@ def stream_data():
                     for trace_log in trace_log_list:
                         key = trace_log["tool"]
                         result_list = trace_log["result"]
+                        if key == 'search_baidu':
+                            self.process_result_list(result_list)
                         self.extra[key] = result_list
 
+    @staticmethod
+    def process_result_list(result_list):
```

Comment:
class ResultProcesser:

    def process(self, key, result_list):
        if key == 'search_baidu':
            rename_fields = {
                'id': 'url',
                'mock_id': 'ref_id',
                'content': 'content',
                'title': 'title',
            }
            renamed_list = []
            for result in result_list:
                renamed_list.append({rename_fields[k]: v for k, v in result.items()})
            return renamed_list
        else:
            raise TypeError(...)